src/main/java/minijava/lexer/Token.java

```diff
@@ -0,0 +1,59 @@
+package minijava.lexer;
+
+/** Lexer token */
+public class Token {
+
+  private final Terminal terminal;
+  private final Location location;
+  private final int content;
+  private final StringTable stringTable;
+
+  public Token(Terminal terminal, Location location, int content, StringTable stringTable) {
```

Comment:
Generally, I like to keep things as hidden as possible (since only the Lexer creates Tokens, the constructor can be package-private, i.e remove the modifier). This applies to a few other places in this PR.